http/http.c

```diff
@@ -834,13 +834,17 @@ http_request_cancel(void * cookie)
  * http_findheader(headers, nheaders, header):
  * Search for ${header} in the ${nheaders} header structures ${headers}.
  * Return a pointer to the associated value or NULL if it is not found.
+ * If ${nheaders} is zero, then ${headers} may be NULL.
```

Comment:
The reason I wanted to add this sentence is because the first sentence is:

> Search for ${header} in the ${nheaders} header structures ${headers}.

If ${headers} is NULL, is it technically "a header structure"?  Or at least, is it a "valid" header structure?
(I'm thinking of the POSIX requirement that `char *` is not NULL.)

As it happens, `http_findheader` can deal with a `headers==NULL` perfectly happily (as long as `nheaders == 0`, which is now enforced with a sanity check).  If we don't want to pass it a NULL, it's certainly possible to modify `http.c` to avoid doing that, but it would involve a bigger set of changes.